tachyon/base/profiler.h

```diff
@@ -0,0 +1,49 @@
+#ifndef TACHYON_BASE_PROFILER_H_
+#define TACHYON_BASE_PROFILER_H_
+
+#include <memory>
+#include <string>
+
+#include "third_party/perfetto/perfetto.h"
+
+#include "tachyon/base/files/file.h"
+
+PERFETTO_DEFINE_CATEGORIES(
+    perfetto::Category("Utils").SetDescription("Base utility functions"),
+    perfetto::Category("Subtask").SetDescription(
+        "Subtask within a bigger task"),
+    perfetto::Category("MSM").SetDescription(
+        "Multi Scalar Multiplication operations"),
+    perfetto::Category("ProofGeneration")
+        .SetDescription("The proof generation process"),
+    perfetto::Category("ProofVerification")
+        .SetDescription("The proof verification process"),
+    perfetto::Category("EvaluationDomain")
+        .SetDescription("Evaluation Domain operations"));
+
+namespace tachyon::base {
+
+class Profiler {
+ public:
+  Profiler() = default;
+  explicit Profiler(std::string_view trace_filepath, size_t max_size_kb);
```

Comment:
I think it would be nice to have a `Option` struct inside `Profiler`

```suggestion
  struct Options {
    base::FilePath path;
    size_t max_size_kb;
  };

  explicit Profiler(const Options& options);
```